### PR TITLE
Ca 18 schedule notifications

### PR DESCRIPTION
**Pull request by JaycobDuffel**

### CODE REVIEW

File: api/management/commands/check_for_notifications.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        for user in User.objects.all():
+            user_schedules = user.get_schedules()
+
+            for schedule in user_schedules:
+                content = schedule.get_content()
+                if not content:
+                    continue
+
+                user_feed = \
+                    UserFeed.objects.filter(user=user, schedule=schedule).order_by('notification_date_time').all()
```
Comment (katiemcgoff) [inline api/management/commands/check_for_notifications.py:24]: I'm wondering if we really need to loop through schedules - I think we could just loop through the user's whole feed, regardless of schedule. We could also probably filter out notifications that are already sent, using your new field, which would yield a significantly smaller list.

File: api/settings.py [hunk 2]
```diff
@@ -175,6 +175,7 @@
 # crontab syntax is: minute hour dayofmonth month dayofweek
 CRONJOBS = [
     ("0 1,13 * * *", "api.cron.cron_create_feed", ">> /tmp/cron_create_feed_output.log"),  # every day at 1 am, 1 pm
+    ("5 * * * *", "api.cron.cron_check_for_notifications", ">> /tmp/cron_check_for_notifications_output.log"),
```
Comment (katiemcgoff) [inline api/settings.py:178]: This timing I believe is actually every hour on the 5-minute, ie: 1:05, 2:05, etc. I'm guessing you were going for every 5 minutes, which would be ("*/5 * * * *"). 

This is a fun tool to experiment with: https://crontab.guru/

File: tests/commands/test_check_for_notifications.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        """ Schedule's default start date is respected."""
+        self.user_a.notification_token = "faketoken"
+        self.user_a.save()
+
+        call_command('create_feed', stdout=self.out)
+        call_command('check_for_notifications', stdout=self.out)
+
+        user_feed = UserFeed.objects.filter(user=self.user_a, schedule=self.schedule_daily)
+
+        self.assertEqual(user_feed.first().notification_sent, False)
```
Comment (katiemcgoff) [inline tests/commands/test_check_for_notifications.py:55]: I'd say an additional test for this file should be that after the notification command runs, that any items in the feed that are in the future have not been notified. I believe the feed currently generates 1 day in the future (it's in settings.py somewhere) but you could change that to two in your test (by inporting settings.py and then changing it to like, 5 or something) to give yourself more wiggle room.

Alternately to that, you could include contents in your user_feed table in your fixture, and then you won't have to call the create_feed command in any of these tests, and you can add data to the table that supports specific tests.

Comment (katiemcgoff) [reply tests/commands/test_check_for_notifications.py:55]: ps - I'm not 100% sure you can change a settings.py setting after the application is running. TBD.

**Review by katiemcgoff — CHANGES_REQUESTED** [review]
changes requested

File: tests/commands/test_check_for_notifications.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        """ check_for_notifications management command runs."""
+
+        call_command('create_feed', stdout=self.out)
+        call_command('check_for_notifications', stdout=self.out)
+
+        self.assertIn('Done.', self.out.getvalue())
+
+    @freeze_time('2020-12-08')
+    def test_changes_notification_sent_to_True_with_valid_token(self):
+        """ Schedule's default start date is respected."""
```
Comment (katiemcgoff) [inline tests/commands/test_check_for_notifications.py:36]: These test descriptions (the comment line) need to reflect the test.

These descriptions are printed when a test fails.

**Review by katiemcgoff — APPROVED** [review]

**Review by DawsonDease — APPROVED** [review]

**Pull request merged**